### PR TITLE
Add automated pipeline testing framework for tw pipelines

### PR DESCRIPTION
**Pull request by debasishbsws**
Introduces a Go-based test runner that auto-generates Melange test configs from declarative YAML test cases, eliminating manual test file duplication.

Key additions:
- pipeline-runner/: Go CLI tool that parses test cases, generates Melange configs, executes tests, and reports failures
- Test case YAML format with expect_pass for positive/negative testing
- Initial test suites for contains-files, docs, emptypackage, and metapackage pipelines
- Debug and verbose logging modes for troubleshooting

Benefits:
- 1:1 mapping: one test case = one package + one expected outcome
- Real package testing against actual Wolfi packages
- Automated validation with detailed failure reporting
- Reduces test maintenance burden from N config files to single test suite definitions

This will not replace the manual test infra we are adding [here](https://github.com/chainguard-dev/tw/pull/271), but complements it for easier configuration for normal everyday package test.

### CODE REVIEW

File: tests/testcases/contains-files-pipeline.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Contains-files pipeline validation tests
```
[the remark this answers is not in the document]
Comment (debasishbsws) [reply tests/testcases/contains-files-pipeline.yaml]: I am thinking about this structure
```
tests/
├── manual/                    # Manual tests (full melange YAML)
│   └── header-check.yaml
├── docs.yaml                  # Suite tests (declarative format)
├── contains-files.yaml
├── emptypackage.yaml
├── metapackage.yaml
├── runner/                    # Test runner tool (infrastructure)
│   ├── main.go
│   ├── go.mod
│   └── pipeline-runner
├── packages/                  # Build artifacts (gitignored)
├── generated/                 # Auto-generated configs (gitignored)
├── README.md
└── .gitignore

```

**Comment by debasishbsws** [comment]
@aborrero This is the approach I have taken which seems the best to me.

This PR reorganizes the test directory into a cleaner structure where every top-level entry has a clear, single purpose:

```
tests/
├── suites/          # Declarative test definitions (consumed by runner)
│   ├── docs.yaml
│   ├── contains-files.yaml
│   ├── emptypackage.yaml
│   └── metapackage.yaml
├── manual/          # Hand-written melange YAML (synthetic packages)
│   └── header-check.yaml
├── runner/          # Test runner Go source (infrastructure)
│   ├── main.go
│   ├── go.mod
│   └── go.sum
├── .out/            # All build artifacts (gitignored)
└── README.md
```

### Why this structure

- **`suites/`** is what you interact with day-to-day. Adding a new pipeline test = dropping a YAML file here. File names map 1:1 to pipelines (`docs.yaml` → `test/tw/docs`). Putting these in a dedicated directory prevents the glob (`*.yaml`) from accidentally picking up unrelated YAML files that might be added to `tests/` in the future.

- **`manual/`** is for edge cases that need synthetic package content (e.g., testing `header-check` with deliberately malformed headers). These are full melange YAML files run directly via `melange build` + `melange test`. Kept separate because they're a fundamentally different kind of test.

- **`runner/`** is infrastructure — you rarely touch it. Lives inside `tests/` because it's test-only plumbing, not a user-facing tool.

- **`.out/`** consolidates what used to be two separate gitignored directories (`generated/` and `packages/`) into one. Hidden by default in `ls`, one entry to gitignore, one `rm -rf` to clean.

### Make targets

```bash
make test-pipelines   # Run everything (manual + suite)
make test-suite       # Run only declarative suite tests
make test-manual      # Run only hand-written manual tests
make test-all         # Run all tests (melange + projects + pipelines)
make clean            # Removes packages/ and tests/.out/
```

### Other changes

- Moved gitignore entry for `tests/.out` to root `.gitignore`, removed `tests/.gitignore`
- Updated runner module path and `generatedSubdir` constant for new paths
- Rewrote `tests/README.md` and `tests/runner/README.md` to document the new structure
- Fixed a pre-existing unbound variable bug in `ldd-check/test.sh` (`$@` with `set -u`)

**Review by aborrero — APPROVED** [review]
LGTM.
